src/psyclone/tests/psyir/transformations/chunk_loop_trans_test.py

```diff
@@ -0,0 +1,401 @@
+                    ScalarType.Precision.SINGLE))
+    ivar = routine.symbol_table.symbol_from_tag(
+            "ivar", symbol_type=DataSymbol, datatype=ScalarType(
+                    ScalarType.Intrinsic.INTEGER,
+                    ScalarType.Precision.SINGLE))
+    binop = BinaryOperation.create(BinaryOperation.Operator.ADD,
+                                   Reference(lvar), Literal("1", INTEGER_TYPE))
+    assign = Assignment.create(Reference(ivar), binop)
+    sched.addchild(assign)
+    parent._variable = lvar
+    chunktrans.validate(parent)
+
+
+def test_blockloop_trans_validate6():
+    '''Test the validate method of ChunkLoopTrans fails when applying it
+    to a loop that has a non-integer loop step'''
+    chunktrans = ChunkLoopTrans()
+    # Construct a Loop with a non-integer step
+    parent = Loop()
+    parent.addchild(Literal("1.0", REAL_DOUBLE_TYPE))
+    parent.addchild(Literal("2560.0", REAL_DOUBLE_TYPE))
+    parent.addchild(Literal("1.1", REAL_DOUBLE_TYPE))
+    parent.addchild(Schedule())
+    with pytest.raises(TransformationError) as excinfo:
+        chunktrans.validate(parent)
+    assert ("Cannot apply a ChunkLoopTrans to a loop with a non-integer "
+           "step size, but a step expression of type 'REAL' was found.") \
+        in str(excinfo.value)
+
+
+def test_blockloop_trans_validate7():
+    '''Test the validate method of ChunkLoopTrans fails when applying it
+    to a loop that contains a CodeBlock'''
+    chunktrans = ChunkLoopTrans()
+    # Construct a Schedule containing a CodeBlock
+    sched = Schedule()
+    sched.addchild(CodeBlock([], CodeBlock.Structure.STATEMENT))
+    parent = Loop()
+    parent.addchild(Literal("1", INTEGER_TYPE))
+    parent.addchild(Literal("512", INTEGER_TYPE))
+    parent.addchild(Literal("1", INTEGER_TYPE))
+    parent.addchild(sched)
+    with pytest.raises(TransformationError) as excinfo:
+        chunktrans.validate(parent)
+    assert ("Cannot apply a ChunkLoopTrans to a loop which contains a "
+            "CodeBlock node.") in str(excinfo.value)
+
+
+def test_blockloop_trans_apply_pos():
+    '''Test the apply method of ChunkLoopTrans for a positive step index'''
+    _, invoke_info = parse(os.path.join(GOCEAN_BASE_PATH, "single_invoke.f90"),
+                           api="gocean1.0")
+    psy = PSyFactory("gocean1.0", distributed_memory=False).\
+        create(invoke_info)
+    schedule = psy.invokes.invoke_list[0].schedule
+    chunktrans = ChunkLoopTrans()
+    chunktrans.apply(schedule.children[0])
+    code = str(psy.gen)
+    correct = \
+        '''DO j_out_var = cu_fld%internal%ystart, cu_fld%internal%ystop, 32
+        j_el_inner = MIN(j_out_var + 32, cu_fld%internal%ystop)
+        DO j = j_out_var, j_el_inner, 1
+          DO i = cu_fld%internal%xstart, cu_fld%internal%xstop, 1
+    '''
+    assert correct in code
+    correct = '''END DO
+        END DO
+      END DO'''
+    assert correct in code
```

Comment:
Can you explicitly assert the the "blocked" (I think now "chunked") annotation has been added to the loop.

Reply:
Added